File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n, 0);
        stack<pair<int, int>> stk;

        for(int i = n-1; i >= 0; i--){
            while(stk.size() && stk.top().first <= temperatures[i]){
                stk.pop();
            }
            if(!stk.size()){
                res[i] = 0;
            }
            else res[i] = stk.top().second - i;
            stk.push({temperatures[i], i});
        }
        return res;         
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        dic = {')': '(', '}': '{', ']': '['}
        stk = []
        for i in s:
            if i in dic.values():
                stk.append(i)
            elif i in dic.keys():
                if not len(stk):
                    return False
                elif stk[-1] != dic[i]:
                    return False
                else:
                    stk.pop()

        return not len(stk)
                